src/components/NetlifyForm/index.jsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * NetlifyForm component consists of server related functionalities that is used posting data and also avoiding captchas.
+ * The base code that is used in both netlifyForm and index are implemented from https://github.com/jon-fm/react-ssg-netlify-forms-demo
+ */
+import React, { useState, useEffect } from "react";
+
+const NetlifyForm = ({
+  formName,
+  beforeSubmit,
+  afterSubmit,
+  contactFormValues,
+  children,
+}) => {
+  // Keep things well-hidden during runtime (and from bots. They are exposed for netlify build bots.
+  const [inNetlifyBuild, setInNetlifyBuild] = useState(true);
+  useEffect(() => {
+    setInNetlifyBuild(false);
+  }, []);
+
+  // Honeypot Input state for "Honeypot" field, which used Honeypot form field.
+  const [honey, setHoney] = useState("");
+  // For application/x-www-form-urlencoded, spaces are to be replaced by +, encodeURIComponent() replacement with an additional replacement of %20 with +. */
+  const formEncodeString = (str) =>
+    encodeURIComponent(str).replace(/%20/g, "+");
+
+  // Converts object to proper form data
+  // name/value pairs are separated by the ampersand (&)
+  // names are separated from values by the equals symbol (=).
+  const encodeData = (obj) =>
+    Object.entries(obj)
+      .map(
+        (pair) => formEncodeString(pair[0]) + "=" + formEncodeString(pair[1])
+      )
+      .join("&");
+
+  // Submit via POST then pass back true
+  // application/x-www-form-urlencoded stands for POST data at HTTP.
+  // body represents HTTP message and send data to server in a query string format.
+  // encodeData used for passing the string
+  const handleSubmit = async () => {
+    return await fetch("/", {
+      method: "POST",
+      headers: { "Content-Type": "application/x-www-form-urlencoded" },
+      body: encodeData({
+        ...contactFormValues,
+        "form-name": formName,
+        info: honey,
+      }),
+    });
+  };
+
+  // checks beforeSubmit, onsubmit, afterSubmit works and has errors
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    if ((beforeSubmit && (await beforeSubmit())) || !beforeSubmit) {
```

Comment:
Why would beforeSubmit not exist or be false? What does beforeSbumit do?

Reply:
Same questions apply to other variables here

Reply:
> Why would beforeSubmit not exist or be false? What does beforeSbumit do?

If beforeSubmit is true, it will not validate form and button will keep its initial state(disabled).